template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/BlackListedLoadBasedAutoscaleRole.java

```diff
@@ -0,0 +1,12 @@
+package com.sequenceiq.cloudbreak.cmtemplate;
+
+public enum BlackListedLoadBasedAutoscaleRole {
+    DATANODE,
+    ZEPPELIN_SERVER,
+    KAFKA_BROKER,
+    NIFI_NODE,
+    NAMENODE,
+    // This is counter intuitive. Gateway and compute host groups have identical GATEWAY roles. So compute host group will be blacklisted initially.
+    // The fallback filter will add only compute host group for load based auto-scale role.
+    GATEWAY
```

Comment:
All of these will likely need to be qualified as ServiceName.RoleName at some point.
e.g. Both HDFS and YARN have a Role called 'GATEWAY' - and I'm guessing other services do as well.

Reply:
Yup, `GATEWAY` wipes off everything from the slate, rendering the service-based logic meaningless.